care/facility/models/patient_consultation.py

```diff
@@ -264,6 +265,25 @@ def save(self, *args, **kwargs):
             self.patient.save(update_fields=["death_datetime"])
         super().save(*args, **kwargs)
 
+    def delete(self, *args):
+        from care.facility.models.bed import ConsultationBed
+        from care.facility.models.patient_investigation import InvestigationValue
+        from care.facility.models.patient_sample import PatientSample
+
+        if InvestigationValue.objects.filter(consultation=self).exists():
+            error = f"Cannot delete PatientConsultation {self.external_id} because they are referenced as `consultation` in InvestigationValue records."
+            raise ValidationError(error)
+
+        if PatientSample.objects.filter(consultation=self).exists():
+            error = f"Cannot delete PatientConsultation {self} because they are referenced as `consultation` in PatientSample records."
+            raise ValidationError(error)
+
+        if ConsultationBed.objects.filter(consultation=self).exists():
+            error = f"Cannot delete PatientConsultation {self} because they are referenced as `consultation` in ConsultationBed records."
+            raise ValidationError(error)
+
+        return super().delete(*args)
```

Comment:
have to keep import inside the function , since they are causing circular imports.

Also since patient consultation is referenced as PROTECTED field , so we don't need to soft delete them but to raise a validation error instead